### PR TITLE
Remove nemsio support

### PR DESCRIPTION
**Pull request by lgannoaa**
**Description**
Removes remaining nemsio support from scripts. Now unnecessary variables will be removed in a follow-up PR.

Refs: #601 

**How Has This Been Tested?**

- [X] Cycled test on WCOSS
  
**Checklist**

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] My changes generate no new warnings

### CODE REVIEW

**Review by aerorahul — APPROVED** [review]
Changes look good to me.
I have not tested this change.
A future PR can remove the source codes that are related to nemsio and the linking of the executables.
This clean-up is greatly appreciated.

File: parm/config/config.base.emc.dyn [hunk 1]
```diff
@@ -281,15 +280,9 @@ export WRITE_DOPOST=".true."
...
-    export NEMSIO_IN=".false."
-    export NETCDF_IN=".true."
-else
-    export SUFFIX=".nemsio"
-    export NEMSIO_IN=".true."
-    export NETCDF_IN=".false."
-fi
+export SUFFIX=".nc"
+export NEMSIO_IN=".false."
+export NETCDF_IN=".true."
```
Comment (WalterKolczynski-NOAA) [inline parm/config/config.base.emc.dyn:285]: Should be able to get rid of these variables entirely now. Find wherever they are used and replace with hard-coded values.

File: scripts/exgdas_enkf_update.sh [hunk 2]
```diff
@@ -89,30 +88,20 @@ INCREMENTS_TO_ZERO=${INCREMENTS_TO_ZERO:-"'NONE'"}
...
-      write_fv3_incr=".false."
-   else
-      write_fv3_incr=".true."
-      WRITE_INCR_ZERO="incvars_to_zero= $INCREMENTS_TO_ZERO,"
-   fi
+LONB_ENKF=${LONB_ENKF:-$($NCLEN $ATMGES_ENSMEAN grid_xt)} # get LONB_ENKF
+LATB_ENKF=${LATB_ENKF:-$($NCLEN $ATMGES_ENSMEAN grid_yt)} # get LATB_ENFK
+LEVS_ENKF=${LEVS_ENKF:-$($NCLEN $ATMGES_ENSMEAN pfull)} # get LEVS_ENFK
+use_gfs_ncio=".true."
+use_gfs_nemsio=".false."
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgdas_enkf_update.sh:95]: These should also be able to be removed now.

File: scripts/exgfs_atmos_postsnd.sh [hunk 3]
```diff
@@ -36,38 +36,20 @@ export INCREMENT=12
 export MAKEBUFR=NO
 export F00FLAG=YES
 export fformat=${OUTPUT_FILE:-netcdf}
-if [ $fformat == "netcdf" ]
- then
 export atmfm="nc"
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_atmos_postsnd.sh:39]: This one too, I think.

File: ush/hpssarch_gen.sh [hunk 4]
```diff
@@ -16,11 +16,7 @@ ARCH_GAUSSIAN=${ARCH_GAUSSIAN:-"YES"}
 ARCH_GAUSSIAN_FHMAX=${ARCH_GAUSSIAN_FHMAX:-36}
 ARCH_GAUSSIAN_FHINC=${ARCH_GAUSSIAN_FHINC:-6}
 SUFFIX=${SUFFIX:-".nc"}
-if [ $SUFFIX = ".nc" ]; then
-  format="netcdf"
-else
-  format="nemsio"
-fi
+format="netcdf"
```
Comment (WalterKolczynski-NOAA) [inline ush/hpssarch_gen.sh:19]: These as well.

File: ush/gfs_post.sh [hunk 5]
```diff
@@ -222,8 +222,7 @@ export LOGSCRIPT=${LOGSCRIPT:-}
 export ENDSCRIPT=${ENDSCRIPT:-}
 export GFSOUT=${GFSOUT:-gfsout}
 export CTLFILE=${CTLFILE:-${NWPROD}/parm/gfs_cntrl.parm}
-#export MODEL_OUT_FORM=${MODEL_OUT_FORM:-binarynemsiompiio}
-export OUTPUT_FILE=${OUTPUT_FILE:-"nemsio"}
+export OUTPUT_FILE=${OUTPUT_FILE:-"netcdf"}
```
Comment (WalterKolczynski-NOAA) [inline ush/gfs_post.sh:225]: And this.

File: ush/gaussian_sfcanl.sh [hunk 6]
```diff
@@ -121,13 +121,8 @@ LATB_SFC=${LATB_SFC:-$LATB_CASE}
...
 LEVSP1=$(($LEVS+1))
-OUTPUT_FILE=${OUTPUT_FILE:-"nemsio"}
-if [ $OUTPUT_FILE = "netcdf" ]; then
-    export NETCDF_OUT=".true."
-else
-    export NETCDF_OUT=".false."
-fi
-
+OUTPUT_FILE=${OUTPUT_FILE:-"netcdf"}
+export NETCDF_OUT=".true."
```
Comment (WalterKolczynski-NOAA) [inline ush/gaussian_sfcanl.sh:125]: And these.

File: ush/forecast_predet.sh [hunk 7]
```diff
@@ -133,11 +133,10 @@ FV3_GFS_predet(){
 
   QUILTING=${QUILTING:-".true."}
   OUTPUT_GRID=${OUTPUT_GRID:-"gaussian_grid"}
-  OUTPUT_FILE=${OUTPUT_FILE:-"nemsio"}
+  OUTPUT_FILE=${OUTPUT_FILE:-"netcdf"}
```
Comment (WalterKolczynski-NOAA) [inline ush/forecast_predet.sh:136]: And this.

File: ush/forecast_predet.sh [hunk 8]
```diff
@@ -133,11 +133,10 @@ FV3_GFS_predet(){
...
   WRITE_NEMSIOFLIP=${WRITE_NEMSIOFLIP:-".true."}
   WRITE_FSYNCFLAG=${WRITE_FSYNCFLAG:-".true."}
-  affix="nemsio"
-  [[ "$OUTPUT_FILE" = "netcdf" ]] && affix="nc"
+  affix="nc"
```
Comment (WalterKolczynski-NOAA) [inline ush/forecast_predet.sh:139]: And this.

File: ush/forecast_postdet.sh [hunk 9]
```diff
@@ -501,14 +501,8 @@ EOF
   JCAP_STP=${JCAP_STP:-$JCAP_CASE}
   LONB_STP=${LONB_STP:-$LONB_CASE}
   LATB_STP=${LATB_STP:-$LATB_CASE}
-
   cd $DATA
-
   affix="nc"
```
Comment (WalterKolczynski-NOAA) [inline ush/forecast_postdet.sh:505]: And this.

File: scripts/exglobal_archive.sh [hunk 10]
```diff
@@ -16,12 +16,7 @@ fi
 # CURRENT CYCLE
 APREFIX="${CDUMP}.t${cyc}z."
 ASUFFIX=${ASUFFIX:-${SUFFIX}}
-
-if [ "${ASUFFIX}" = ".nc" ]; then
-   format="netcdf"
-else
-   format="nemsio"
-fi
+format="netcdf"
```
Comment (WalterKolczynski-NOAA) [inline scripts/exglobal_archive.sh:19]: And this.

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
We can remove a lot of variables that aren't really needed anymore now that only netcdf is supported. For those that are used in namelists or filenames, you'll need to replace them with the hard-coded value.

**Comment by lgannoaa** [comment]
> We can remove a lot of variables that aren't really needed anymore now that only netcdf is supported. For those that are used in namelists or filenames, you'll need to replace them with the hard-coded value.

@WalterKolczynski-NOAA, Would you please check again these change request? It save time and resource to make modification once for all.

**Comment by WalterKolczynski-NOAA** [comment]
> > We can remove a lot of variables that aren't really needed anymore now that only netcdf is supported. For those that are used in namelists or filenames, you'll need to replace them with the hard-coded value.
> 
> @WalterKolczynski-NOAA, Would you please check again these change request? It save time and resource to make modification once for all.

I... don't know what you're asking for here.

File: scripts/exgdas_enkf_update.sh (hunk 2 above)
Comment (lgannoaa) [reply scripts/exgdas_enkf_update.sh:95]: (1) The use_gfs_nemsio is part of the enkf.nml namelist. It should not be changed. 
(2) The variable use_gfs_nemsio is only used locally.
Therefore, recommend no change here.

File: scripts/exglobal_archive.sh (hunk 10 above)
Comment (lgannoaa) [reply scripts/exglobal_archive.sh:19]: Variable format is used locally. Therefore it should be ok not to change it.

File: ush/forecast_postdet.sh (hunk 9 above)
Comment (lgannoaa) [reply ush/forecast_postdet.sh:505]: Variable affix is used locally. Therefore, no need to modify.

Comment (WalterKolczynski-NOAA) [reply ush/forecast_postdet.sh:505]: The point is I want you to replace `$affix` with `"nc"` everywhere it appears. Similarly for the others.

**Comment by WalterKolczynski-NOAA** [comment]
Any variables that are no longer needed for switching to nemsio should be removed, and any variables that no longer change (like `affix`) should be replaced with their static value. There may be more than I have spotted during review. Please examine the scripts.

File: scripts/exgdas_enkf_update.sh (hunk 2 above)
Comment (aerorahul) [reply scripts/exgdas_enkf_update.sh:95]: There is no need for these variables. So they could/should be removed.

File: ush/forecast_postdet.sh (hunk 9 above)
Comment (aerorahul) [reply ush/forecast_postdet.sh:505]: since there is not other `affix`, this variable is no longer needed and should be removed as the reviewer suggests.

**Comment by aerorahul** [comment]
@lgannoaa would like to address the reviewer comments in a follow-up PR.
Issue #601 will be left opened until the reviewer comments are resolved in the follow-up PR.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Approved with understanding the now unnecessary variables will be removed in a follow-up PR.

**Pull request merged**